kernel/cClosure.ml

```diff
@@ -647,6 +659,10 @@ let rec zip m stk =
     | Zupdate(rf)::s ->
       (** The stack contains [Zupdate] marks only if in sharing mode *)
         zip (update ~share:true rf m.norm m.term) s
+    | Zprimitive(_op,c,rargs,kargs)::s ->
+      let args = List.rev_append rargs (m::List.map snd kargs) in
+      let f = {norm = Red;term = FFlex (ConstKey c)} in
+      zip {norm=neutr m.norm; term = FApp (f, Array.of_list args)} s
```

Comment:
Something fishy about the `norm` field here. How can `f` be `Red` and the result potentially whnf?

Reply:
It is like a blocked fixpoint (if we had global fixpoints): `f` could be unfolded, but if the argument is not a constructor (and not reducible), we don't want to reduce `f`.